test/e2e/framework/network_pods.go

```diff
@@ -404,6 +469,38 @@ func (np *NetworkPod) buildLatencyServerPod() {
 	np.AwaitReady()
 }
 
+// create a test pod inside the current test namespace on the specified cluster.
+// The pod will use the image specified and run command specified.
+func (np *NetworkPod) buildCustomPod() {
+	customPod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "custom",
+			Labels: map[string]string{
+				TestAppLabel: "custom",
+			},
+		},
+		Spec: v1.PodSpec{
+			Affinity:      nodeAffinity(np.Config.Scheduling),
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Name:            np.Config.ContainerName,
+					Image:           np.Config.ImageName,
+					ImagePullPolicy: v1.PullAlways,
+					Command:         np.Config.Command,
+				},
+			},
+			Tolerations: []v1.Toleration{{Operator: v1.TolerationOpExists}},
+		},
+	}
+
+	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
+	var err error
+	np.Pod, err = pc.Create(&customPod)
+	Expect(err).NotTo(HaveOccurred())
+	np.AwaitReady()
```

Comment:
maybe add some newlines for readability...

```suggestion
	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
	
	var err error
	np.Pod, err = pc.Create(&customPod)
	Expect(err).NotTo(HaveOccurred())
	
	np.AwaitReady()
```